embedder/embedder.go

```diff
@@ -164,13 +164,23 @@ const (
 	PointerSignalKindScroll PointerSignalKind = C.kFlutterPointerSignalKindScroll
 )
 
+// PointerDeviceKind corresponds to the C.enum describing device kind of the mouse pointer.
+type PointerDeviceKind int32
+
+// Values representing the pointer signal kind.
+const (
+	PointerDeviceKindMouse PointerDeviceKind = C.kFlutterPointerDeviceKindMouse
+	PointerDeviceKindTouch PointerDeviceKind = C.kFlutterPointerDeviceKindTouch
+)
+
 // PointerEvent represents the position and phase of the mouse at a given time.
 type PointerEvent struct {
 	Phase        PointerPhase
 	Timestamp    int64
 	X            float64
 	Y            float64
 	SignalKind   PointerSignalKind
+	DeviceKind   PointerDeviceKind
```

Comment:
DeviceKind must be set for all PointerEvent not just in sendPointerEventButton
[Doc](https://github.com/flutter/engine/blob/8812781d912f85db71d64a2ec865782d3a2e92bf/shell/platform/embedder/embedder.h#L353-L358)